pkg/dogstatsd/server.go

```diff
@@ -446,10 +455,12 @@ func (s *Server) ServerlessFlush() {
 	for _, w := range s.workers {
 		w.flush()
 	}
+
+	start := time.Now()
+
 	// flush the aggregator to have the serializer/forwarder send data to the backend.
-	agg := s.demultiplexer.Aggregator()
-	agg.ServerlessFlush <- true
-	<-agg.ServerlessFlushDone
+	// We add 10 seconds to the interval to ensure that we're getting the whole sketches bucket
+	s.demultiplexer.ForceFlushToSerializer(start.Add(time.Second*10), true)
```

Comment:
We used ServerlessFlushDone to make sure the flush was synchronous, is `ForceFlushToSerializer` blocking?

Reply:
Yep I've taken care of this:

Previous to my change, this `ServerlessFlushDone` was filled [here in the aggregator](https://github.com/DataDog/datadog-agent/pull/10532/files#diff-0940f1dc68973941178797f388ffb31d3318abeefa87b7f0b66d575915a0e575L834), it was already using `waitForSerializer` true so I _think_ it was actually redundant. 

With the change in this pull-request, setting `waitForSerializer` parameter to `true` let the flushing mechanism that we are waiting on a channel for the end of the flush: 
  1. here [we trigger a flush](https://github.com/DataDog/datadog-agent/pull/10532/files#diff-29b35a8b2b30e7c024255d7c86cb52e8b6f7b5401f441c7ad005a9211ec4dc39R68-R76) and you can see that this `flushTrigger` is accompanied with a channel 
  2. this channel will receive something only once the flush is over, [here](https://github.com/DataDog/datadog-agent/pull/10532/files#diff-1bc4614e41a3b27abc34f3f211edc7e59151a6180117f55d07de8f4b99812a40R102-R106)
  3. and we are correctly [waiting for this message](https://github.com/DataDog/datadog-agent/pull/10532/files#diff-29b35a8b2b30e7c024255d7c86cb52e8b6f7b5401f441c7ad005a9211ec4dc39R78-R79)